storage/remote/client.go

```diff
@@ -151,7 +151,17 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	if span := opentracing.SpanFromContext(ctx); span != nil {
+	var span opentracing.Span
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
```

Comment:
have you considered simply using a middleware like https://github.com/opentracing-contrib/go-stdlib ?

Reply:
thank you for your review, I will look into that.